### ISSUE TITLE
Visit by days since last visit seems wrong

### ISSUE THREAD

**Issue by milexjaro** [piwik/piwik#9839]
I have an issue regarding to the title, when a visitor visited a page at 24th Feb 2016 and the last visit was at 18th Feb 2016, why is the visitors->engagement tab not showing the 'days since last visit' table correctly? It's supposed to have a number in '6 days' row, right?
![capture](https://cloud.githubusercontent.com/assets/11173209/13279724/62ac3240-db0c-11e5-80da-48696c708809.JPG)
![capture3](https://cloud.githubusercontent.com/assets/11173209/13279729/6a07a3c6-db0c-11e5-82fa-e3cb74c712bd.JPG)
![capture2](https://cloud.githubusercontent.com/assets/11173209/13279732/6d727338-db0c-11e5-9c03-10ff1b2699e4.JPG)

**Comment by tsteur** [comment]
Thx for creating the issue, this looks like a bug indeed! Maybe it's related to date ranges. Can you see "6 days" when selecting the "Day Feb 24th"?

**Comment by milexjaro** [comment]
![1](https://cloud.githubusercontent.com/assets/11173209/13316087/dcfef8f0-dbe0-11e5-9084-6e82e76c39c0.JPG)
![2](https://cloud.githubusercontent.com/assets/11173209/13316088/de5c0698-dbe0-11e5-9e0c-dfb015c26a46.JPG)

Thanks for replying, and here are the screenshots, and I still can't see "6 days"  when selected the "Day Feb 24th"

**Comment by tsteur** [comment]
Sorry but one more question: Have you seen this problem also for other visitors? It's possible that maybe just the database had once a connection problem or so and one update was not written on a tracking request. Maybe it works for other visitors?

**Comment by mattab** [comment]
it is possible that the user could connect to your website from several devices. As you are using [User ID](http://piwik.org/docs/user-id/) feature, the visitor is correctly recognised, but the "Visits by days since last visit" relies on First Party Cookies ([FAQ](https://piwik.org/faq/general/faq_18254/)) so if your user has used different devices or browsers, it could explain that "Days since last visit" is not correctly set.

**Comment by tsteur** [comment]
@mattab I reckon the reason visits since last days is not calculated server side is re archiving performance?

**Comment by mattab** [comment]
Yes, it would be slow because we don't have the data available in easy to query format. Unless we would select the past historical visits which would be too costly. In the future this could be improved when we store users in a user table (eg `log_user`, which will be needed for several important features such as Life Time Revenue LTV https://github.com/piwik/piwik/issues/2750  or multi channel attribution modelling #6064

**Comment by milexjaro** [comment]
Hi, sorry for the late reply, I found this issue is experienced by other visitors

**Comment by mattab** [comment]
We haven't been able to reproduce unfortunately. Maybe someone else has experienced this issue?

**Comment by huang0808** [comment]
@milexjaro @tsteur @mattab this issue I met too, seems the wrong show in Visits by Days Since Last Visit. beacuse in 365+ days, i did not put tracking code in website, but 365+ days show a value. x

**Comment by ansermgw** [comment]
@mattab When u do a setUserId it delete First Party Cookies or at least change the visitorId. Could this process explain the fact that the engagement report are wrong ?

**Comment by pardvm** [comment]
I also am experiencing the effect described here: the vast majority of visits fall in the "0 days since last visit". However, I can see that it is not true in some cases and, also, that it should be visits in others segements but it isn't.

My matomo version is 3.13.5

Is there anybody out there who has any explanation for this?

CC: @eramirezprotec

**Comment by mattab** [comment]
This would be the case if you disabled tracking cookies or your users don't support cookies in general? Learn more: https://matomo.org/faq/general/faq_156/

**Comment by pardvm** [comment]
@mattab,
I don't think that the effect is being caused by disabling of cookies in any form: I have 50 sites from several places around the world showing the same effect and it is not likely that 99% of the visitors of all of them are in the "new visits" and "0 days since last visit" segments.

I'll continue observing the case but... I'm really stuck with this.

**Comment by mattab** [comment]
@pardvm could you post some screenshot of the buggy reports for various websites in your Matomo? Thanks

**Comment by pardvm** [comment]
@mattab,
Sure. Here you are:

### **Colombia**
![Colombia](https://user-images.githubusercontent.com/25912392/98953412-f4104e80-24fc-11eb-9444-62f08e311a93.png)

### **Malaysia**
![Malaysia](https://user-images.githubusercontent.com/25912392/98953429-f8d50280-24fc-11eb-8646-ba7758c7035f.png)

### **Portugal**
![Portugal](https://user-images.githubusercontent.com/25912392/98953442-fa9ec600-24fc-11eb-82f3-71044ebe0b04.png)

### **Spain**
![Spain](https://user-images.githubusercontent.com/25912392/98953450-fd012000-24fc-11eb-8be0-a8482a29429d.png)

### **Uruguay**
![Uruguay](https://user-images.githubusercontent.com/25912392/98953460-fecae380-24fc-11eb-96e4-aaf955a30ed9.png)
